File: Final.client/src/models/Keep.js
import { Profile } from "./Account"

export class Keep{
    constructor(data)
    {
        this.id = data.id
        this.creatorId = data.creatorId
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.views = data.views
        this.kept = data.kept
        this.creator = data.creator ? new Profile(data.creator) : null
    }
}

export class KeepInVault extends Keep{
    constructor(data)
    {
        super(data)
        this.vaultKeepId = data.vaultKeepId
        this.isPrivate = data.isPrivate
    }
}

File: Final/Models/Keep.cs
namespace Final.Models;

public class Keep
{
    public int id { get; set; }
    public string creatorId { get; set; }
    public string name { get; set; }
    public string description { get; set; }
    public string img { get; set; }
    public int views { get; set; }
    public int kept { get; set; }
    public Account creator { get; set; }
}

public class KeepViewModel : Keep
{
    public int vaultKeepId { get; set; }
    public Boolean isPrivate { get; set; }
}